tests/basic/distribute/phase1-migration-fops.t

```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
+
+. $(dirname $0)/../../include.rc
+. $(dirname $0)/../../volume.rc
+. $(dirname $0)/../../fallocate.rc
+
+#This tests checks if the fops do FOP2 on the dst
+
+cleanup;
+
+TEST glusterd
+TEST pidof glusterd
+TEST $CLI volume create $V0 $H0:$B0/${V0}{0,1}
+TEST $CLI volume set $V0 performance.quick-read off
+TEST $CLI volume set $V0 performance.io-cache off
+TEST $CLI volume set $V0 performance.write-behind off
+TEST $CLI volume set $V0 performance.stat-prefetch off
+TEST $CLI volume set $V0 performance.read-ahead off
+TEST $CLI volume set $V0 delay-gen posix
+TEST $CLI volume set $V0 delay-gen.delay-duration 10000000 #10 seconds
+TEST $CLI volume set $V0 delay-gen.delay-percentage 100
+TEST $CLI volume set $V0 delay-gen.enable read
+TEST $CLI volume start $V0
+TEST $GFS --volfile-id=/$V0 --volfile-server=$H0 $M0
+TEST $GFS --volfile-id=/$V0 --volfile-server=$H0 $M1
+cd $M0
+rebalanced_file_after=""
+for i in {1..10};
+do
+    echo abc > $i;
+    mv $i ${i}.txt #Hoping at least one file will hash to another subvolume after rename
+    if [ -f $B0/${V0}0/${i}.txt ] && [ -f $B0/${V0}1/${i}.txt ];
+    then
+        rebalanced_file=${i}
+        rebalanced_file_after=${i}.txt
+        setfattr -n trusted.distribute.migrate-data -v force ${i}.txt &
+        break;
+    fi
+done
+
+TEST $CLI volume profile $V0 start
+TEST [ ! -z $rebalanced_file ];
+sleep 1 #Assuming rebalance does read fop in a second
+echo abc >> $M1/$rebalanced_file_after
+EXPECT '8' stat -c %s $M1/$rebalanced_file_after
+TEST truncate -s 0 $M1/$rebalanced_file_after
+EXPECT '0' stat -c %s $M1/$rebalanced_file_after
+#TEST chmod +x $M1/$rebalanced_file_after
```

Comment:
Uncommenting this line leads to failure. Phase-1 attrs are reset after it. This happens without my changes too. Wondering if you guys know why this happens.

```
-rw-r-Sr-T. 2 root root 4 Apr 11 05:43 /d/backends/patchy0/1.txt
---------T. 2 root root 4 Apr 11 05:43 /d/backends/patchy1/1.txt
ok  18 [   1235/     11] <  45> 'chmod +x /mnt/glusterfs/1/1.txt'
ok  19 [    163/     14] <  47> 'fallocate -l 1M /mnt/glusterfs/1/1.txt'
-rwxr-xr-x. 2 root root 1048576 Apr 11 05:43 /d/backends/patchy0/1.txt
---------T. 2 root root       4 Apr 11 05:43 /d/backends/patchy1/1.txt
ok  20 [    202/     16] <  50> 'fallocate -p -o 512k -l 128k /mnt/glusterfs/1/1.txt'
-rwxr-xr-x. 2 root root 1048576 Apr 11 05:43 /d/backends/patchy0/1.txt
---------T. 2 root root       4 Apr 11 05:43 /d/backends/patchy1/1.txt
ok  21 [    204/     11] <  52> 'truncate -s 0 /mnt/glusterfs/1/1.txt'
-rwxr-xr-x. 2 root root 0 Apr 11 05:43 /d/backends/patchy0/1.txt
---------T. 2 root root 4 Apr 11 05:43 /d/backends/patchy1/1.txt
ok  22 [    203/     32] <  54> '0 stat -c %s /mnt/glusterfs/1/1.txt'
-rwxr-xr-x. 2 root root 4 Apr 11 05:43 /d/backends/patchy0/1.txt
---------T. 2 root root 4 Apr 11 05:43 /d/backends/patchy1/1.txt
ok  23 [    203/     32] <  57> '4 stat -c %s /mnt/glusterfs/1/1.txt'
```

Reply:
@mohit84 @BarakSason @tshacked Is this a day-1 bug?

Reply:
It is possible that this particular thing is a day-1 issue (considering the phase1 flags are managed by chmod itself, along with xattrs).

Reply:
Okay, will leave it as is in that case for now.